---
title: "BacPoSSR Tutorial"
author: "Izumi Ando"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Tutorial_BacPoSSR}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
options(repos = c(CRAN = "https://cran.r-project.org"))

# Attach the package
library(BacPoSSR)
```

## Introduction

`BacPoSSR` is a tool to filter feature data to reduce population structure signal before conducting  bacterial GWAS / machine learning predictive studies.

To install and attach `BacPoSSR` to your current work session, run the following lines in your R console.
```{r, eval = FALSE}
# install.packages("devtools")
library("devtools")
devtools::install_github("izumiando/BacPoSSR", build_vignettes = TRUE)
library("BacPoSSR")
```

To view the functions available in this package, run the following line in your R console.
```{r, eval = FALSE}
ls("package:BacPoSSR")
```
`
Sample datasets are provided in the `inst/extdata` subdirectory. 
You can load them and use them in your R environment using the following lines of code. An example of how this can be used is included in the Functionality section of this tutorial.
As of the most recent update to this package on Nov 27th, 2024, the sample datasets available are dummy data.
```{r}
# replace "file.csv" with the file you want to load
# filePath <- system.file("extdata", "file.csv", package = "BacPoSSR")
# loadedData <- read.csv(filePath, header = TRUE)
```

To to run the Shiny app for this package, run the following line in your R console.
```{r, eval = FALSE}
runBacPoSSR()
```


## Functionality
In this section, we will go through how to use the functions with the sample dataset provided in the package.

### Step 0 : Load Data
We will first start by loading the data into our R environment and formatting it to comply with the input requirements of the functions.

```{r, eval = FALSE}
# getting the file paths for the sample data
pathFeatureMatrix <- system.file("extdata", 
                                 "sampleFeatureMatrix.csv", 
                                 package = "BacPoSSR")
pathPhenotypeData <- system.file("extdata", 
                                 "samplePhenotypeData.csv", 
                                 package = "BacPoSSR")
pathGroupAssignments <- system.file("extdata", 
                                    "sampleGroupAssignments.csv", 
                                    package = "BacPoSSR")

# load the data from the file paths
sampleFeatureMatrix <- read.csv(pathFeatureMatrix, header = TRUE)
samplePhenotypeData <- read.csv(pathPhenotypeData, header = TRUE)
sampleGroupAssignments <- read.csv(pathGroupAssignments, header = TRUE)

# format the data objects so that 
# the column with sample names are assigned as row names

# assigning first column as row names
row.names(sampleFeatureMatrix) <- sampleFeatureMatrix[, 1] 
# removing redundant first column
sampleFeatureMatrix <- sampleFeatureMatrix[, -1]

sampleNames <- samplePhenotypeData[, 1]
samplePhenotypeData <- as.data.frame(samplePhenotypeData[, -1])
row.names(samplePhenotypeData) <- sampleNames
colnames(samplePhenotypeData) <- c("phenotypes")

sampleGroupAssignments <- as.data.frame(sampleGroupAssignments[, -1])
row.names(sampleGroupAssignments) <- sampleNames
colnames(sampleGroupAssignments) <- c("groups")
```

### Step 1 : Multiple Component Analysis (MCA) on Unfiltered Dataset
Now that the data is ready to use in our R environment, we will first examine the population structure of the unfiltered dataset by plotting the MCA. (Note: The sample dataset is a dummy data so the plot will not be representative of what real data would look like. In a dataset with strong population signal, we would expect to see clusters of points with the same color.)

```{r, eval = FALSE}
sampleUnfilteredMCA <-
  BacPoSSR::plotMultCompAnalysis(featureMatrix = sampleFeatureMatrix,
                                 groups = sampleGroupAssignments,
                                 saveTo = NULL,
                                 title = "Sample Unfiltered Dataaset")
```

### Step 2 : Filter Data
Next, we will filter our dataset using the two methods provided in this package. For this first run, we will use the default threshold parameters.

```{r, eval = FALSE}
sampleFilteredMatrixPA <-
  BacPoSSR::filterProcAnalysis(featureMatrix = sampleFeatureMatrix,
                               phenotypes = samplePhenotypeData,
                               threshold = 0.1)

sampleFilteredMatrixGR <-
  BacPoSSR::filterGenRelatedness(featureMatrix = sampleFeatureMatrix,
                                 grm = NULL,
                                 thresholdP = 0.05)
sampleFilteredMatrixGR <- sampleFilteredMatrixGR[[1]]
```


### Step 3 : MCA on Filtered Dataset
We will examine the efficacy of the filtering done in Step 2 by plotting the MCA of the filtered feature matrices. If the filtering is effective, the color segregated clusters seen in Step 1 should be reduced and we should see the colored points over lapping with others. If the clustering persists, try either re-filtering with a tighter threshold or try another filtering method. (Note: again, since the sample data used in this tutorial is dummy data, the described phenomena may not be observed.)
```{r, eval = FALSE}
# plotting the sample dataset post filtering with filterProcAnalysis 
# with a threshold of 0.1
sampleFilteredProcAnalysisMCA <-
  BacPoSSR::plotMultCompAnalysis(featureMatrix = sampleFilteredMatrixPA,
                                 groups = sampleGroupAssignments,
                                 saveTo = NULL,
                                 title = "Sample Filtered Dataaset, PA, 0.1")
# plotting the sample dataset post filtering with filterGenRelatedness 
# with a p-value threshold of 0.05
sampleFilteredGenRelMCA <-
  BacPoSSR::plotMultCompAnalysis(featureMatrix = sampleFilteredMatrixGR,
                                 groups = sampleGroupAssignments,
                                 saveTo = NULL,
                                 title = "Sample Filtered Dataaset, GR, 0.05")
```

## Referencing this Package

Please use the following citation to reference this package.
<br>

- Ando, I. (2024) BacPoSSR: Filtering Tool to Reduce Population Structure Signal in Bacterial Genotype-Phenotype Analysis Data. Unpublished. https://github.com/izumiando/BacPoSSR

## References for the Development of this Package

- Chang W, Cheng J, Allaire J, Sievert C, Schloerke B, Xie Y, Allen J, McPherson J, Dipert A, Borges B (2024). _shiny: Web Application Framework for R_. R package version 1.9.1, <https://CRAN.R-project.org/package=shiny>.

- Karlsen, S. T., Rau, M. H., Sánchez, B. J., Jensen, K., & Zeidan, A. A. (2023). From genotype to phenotype: Computational approaches for inferring microbial traits relevant to the food industry. FEMS Microbiology Reviews, 47(4), fuad030. https://doi.org/10.1093/femsre/fuad030

- Lê S, Josse J, Husson F (2008). “FactoMineR: A Package for Multivariate Analysis.” Journal of Statistical Software, 25(1), 1–18. doi:10.18637/jss.v025.i01.

- Lees, J. A., Galardini, M., Bentley, S. D., Weiser, J. N., & Corander, J. (2018). pyseer: A comprehensive tool for microbial pangenome-wide association studies. Bioinformatics, 34(24), 4310–4312. https://doi.org/10.1093/bioinformatics/bty539

- Oksanen J, Simpson G, Blanchet F, Kindt R, Legendre P, Minchin P, O'Hara R, Solymos P, Stevens M, Szoecs E, Wagner H, Barbour M, Bedward M, Bolker B, Borcard D, Carvalho G, Chirico M, De Caceres M, Durand S, Evangelista H, FitzJohn R, Friendly M, Furneaux B, Hannigan G, Hill M, Lahti L, McGlinn D, Ouellette M, Ribeiro Cunha E, Smith T, Stier A, Ter Braak C, Weedon J (2024). _vegan: Community Ecology Package_. R package version 2.6-8, <https://CRAN.R-project.org/package=vegan>.

- R Core Team (2024). _R: A Language and Environment for Statistical Computing_. R Foundation for Statistical Computing, Vienna, Austria. <https://www.R-project.org/>.

- Sievert C, Cheng J, Aden-Buie G (2024). _bslib: Custom 'Bootstrap' 'Sass' Themes for 'shiny' and 'rmarkdown'_. R package version 0.8.0, <https://CRAN.R-project.org/package=bslib>.

- Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York. ISBN 978-3-319-24277-4, https://ggplot2.tidyverse.org.


```{r}
# sessionInfo()
```
