<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Izumi Ando" />


<title>BacPoSSR Tutorial</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">BacPoSSR Tutorial</h1>
<h4 class="author">Izumi Ando</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p><code>BacPoSSR</code> is a tool to filter feature data to reduce
population structure signal before conducting bacterial GWAS / machine
learning predictive studies.</p>
<p>To install and attach <code>BacPoSSR</code> to your current work
session, run the following lines in your R console.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># install.packages(&quot;devtools&quot;)</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;izumiando/BacPoSSR&quot;</span>, <span class="at">build_vignettes =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;BacPoSSR&quot;</span>)</span></code></pre></div>
<p>To view the functions available in this package, run the following
line in your R console.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">ls</span>(<span class="st">&quot;package:BacPoSSR&quot;</span>)</span></code></pre></div>
<p><code>Sample datasets are provided in the</code>inst/extdata`
subdirectory. You can load them and use them in your R environment using
the following lines of code. An example of how this can be used is
included in the Functionality section of this tutorial. As of the most
recent update to this package on Nov 27th, 2024, the sample datasets
available are dummy data.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="co"># replace &quot;file.csv&quot; with the file you want to load</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a><span class="co"># filePath &lt;- system.file(&quot;extdata&quot;, &quot;file.csv&quot;, package = &quot;BacPoSSR&quot;)</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="co"># loadedData &lt;- read.csv(filePath, header = TRUE)</span></span></code></pre></div>
<p>To to run the Shiny app for this package, run the following line in
your R console.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">runBacPoSSR</span>()</span></code></pre></div>
</div>
<div id="functionality" class="section level2">
<h2>Functionality</h2>
<p>In this section, we will go through how to use the functions with the
sample dataset provided in the package.</p>
<div id="step-0-load-data" class="section level3">
<h3>Step 0 : Load Data</h3>
<p>We will first start by loading the data into our R environment and
formatting it to comply with the input requirements of the
functions.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="co"># getting the file paths for the sample data</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>pathFeatureMatrix <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>                                 <span class="st">&quot;sampleFeatureMatrix.csv&quot;</span>, </span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>                                 <span class="at">package =</span> <span class="st">&quot;BacPoSSR&quot;</span>)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>pathPhenotypeData <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>                                 <span class="st">&quot;samplePhenotypeData.csv&quot;</span>, </span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>                                 <span class="at">package =</span> <span class="st">&quot;BacPoSSR&quot;</span>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>pathGroupAssignments <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>                                    <span class="st">&quot;sampleGroupAssignments.csv&quot;</span>, </span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>                                    <span class="at">package =</span> <span class="st">&quot;BacPoSSR&quot;</span>)</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="co"># load the data from the file paths</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>sampleFeatureMatrix <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(pathFeatureMatrix, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>samplePhenotypeData <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(pathPhenotypeData, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>sampleGroupAssignments <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(pathGroupAssignments, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a></span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="co"># format the data objects so that </span></span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="co"># the column with sample names are assigned as row names</span></span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a></span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="co"># assigning first column as row names</span></span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="fu">row.names</span>(sampleFeatureMatrix) <span class="ot">&lt;-</span> sampleFeatureMatrix[, <span class="dv">1</span>] </span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="co"># removing redundant first column</span></span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a>sampleFeatureMatrix <span class="ot">&lt;-</span> sampleFeatureMatrix[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a></span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a><span class="fu">row.names</span>(samplePhenotypeData) <span class="ot">&lt;-</span> samplePhenotypeData[, <span class="dv">1</span>]</span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a>samplePhenotypeData <span class="ot">&lt;-</span> samplePhenotypeData[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a><span class="co"># making sure object stays a data frame</span></span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a>samplePhenotypeData <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(samplePhenotypeData)</span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a></span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a><span class="fu">row.names</span>(sampleGroupAssignments) <span class="ot">&lt;-</span> sampleGroupAssignments[, <span class="dv">1</span>]</span>
<span id="cb5-31"><a href="#cb5-31" tabindex="-1"></a>sampleGroupAssignments <span class="ot">&lt;-</span> sampleGroupAssignments[, <span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb5-32"><a href="#cb5-32" tabindex="-1"></a>sampleGroupAssignments <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(sampleGroupAssignments)</span></code></pre></div>
</div>
<div id="step-1-multiple-component-analysis-mca-on-unfiltered-dataset" class="section level3">
<h3>Step 1 : Multiple Component Analysis (MCA) on Unfiltered
Dataset</h3>
<p>Now that the data is ready to use in our R environment, we will first
examine the population structure of the unfiltered dataset by plotting
the MCA. (Note: The sample dataset is a dummy data so the plot will not
be representative of what real data would look like. In a dataset with
strong population signal, we would expect to see clusters of points with
the same color.)</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>sampleUnfilteredMCA <span class="ot">&lt;-</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  BacPoSSR<span class="sc">::</span><span class="fu">plotMultCompAnalysis</span>(<span class="at">featureMatrix =</span> sampleFeatureMatrix,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>                                 <span class="at">groups =</span> sampleGroupAssignments,</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>                                 <span class="at">saveTo =</span> <span class="cn">NULL</span>,</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">&quot;Sample Unfiltered Dataaset&quot;</span>)</span></code></pre></div>
</div>
<div id="step-2-filter-data" class="section level3">
<h3>Step 2 : Filter Data</h3>
<p>Next, we will filter our dataset using the two methods provided in
this package. For this first run, we will use the default threshold
parameters.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>sampleFilteredMatrixPA <span class="ot">&lt;-</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  BacPoSSR<span class="sc">::</span><span class="fu">filterProcAnalysis</span>(<span class="at">featureMatrix =</span> sampleFeatureMatrix,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>                               <span class="at">phenotypes =</span> samplePhenotypeData,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                               <span class="at">threshold =</span> <span class="fl">0.1</span>)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>sampleFilteredMatrixGR <span class="ot">&lt;-</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>  BacPoSSR<span class="sc">::</span><span class="fu">filterGenRelatedness</span>(<span class="at">featureMatrix =</span> sampleFeatureMatrix,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>                                 <span class="at">grm =</span> <span class="cn">NULL</span>,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                                 <span class="at">thresholdP =</span> <span class="fl">0.05</span>)</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>sampleFilteredMatrixGR <span class="ot">&lt;-</span> sampleFilteredMatrixGR[[<span class="dv">1</span>]]</span></code></pre></div>
</div>
<div id="step-3-mca-on-filtered-dataset" class="section level3">
<h3>Step 3 : MCA on Filtered Dataset</h3>
<p>We will examine the efficacy of the filtering done in Step 2 by
plotting the MCA of the filtered feature matrices. If the filtering is
effective, the color segregated clusters seen in Step 1 should be
reduced and we should see the colored points over lapping with others.
If the clustering persists, try either re-filtering with a tighter
threshold or try another filtering method. (Note: again, since the
sample data used in this tutorial is dummy data, the described phenomena
may not be observed.)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a><span class="co"># plotting the sample dataset post filtering with filterProcAnalysis </span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a><span class="co"># with a threshold of 0.1</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>sampleFilteredProcAnalysisMCA <span class="ot">&lt;-</span></span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>  BacPoSSR<span class="sc">::</span><span class="fu">plotMultCompAnalysis</span>(<span class="at">featureMatrix =</span> sampleFilteredMatrixPA,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>                                 <span class="at">groups =</span> sampleGroupAssignments,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>                                 <span class="at">saveTo =</span> <span class="cn">NULL</span>,</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">&quot;Sample Filtered Dataaset, PA, 0.1&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="co"># plotting the sample dataset post filtering with filterGenRelatedness </span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="co"># with a p-value threshold of 0.05</span></span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>sampleFilteredGenRelMCA <span class="ot">&lt;-</span></span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  BacPoSSR<span class="sc">::</span><span class="fu">plotMultCompAnalysis</span>(<span class="at">featureMatrix =</span> sampleFilteredMatrixGR,</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>                                 <span class="at">groups =</span> sampleGroupAssignments,</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>                                 <span class="at">saveTo =</span> <span class="cn">NULL</span>,</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>                                 <span class="at">title =</span> <span class="st">&quot;Sample Filtered Dataaset, GR, 0.05&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="referencing-this-package" class="section level2">
<h2>Referencing this Package</h2>
<p>Please use the following citation to reference this package. <br></p>
<ul>
<li>Ando, I. (2024) BacPoSSR: Filtering Tool to Reduce Population
Structure Signal in Bacterial Genotype-Phenotype Analysis Data.
Unpublished. <a href="https://github.com/izumiando/BacPoSSR" class="uri">https://github.com/izumiando/BacPoSSR</a></li>
</ul>
</div>
<div id="references-for-the-development-of-this-package" class="section level2">
<h2>References for the Development of this Package</h2>
<ul>
<li><p>Chang W, Cheng J, Allaire J, Sievert C, Schloerke B, Xie Y, Allen
J, McPherson J, Dipert A, Borges B (2024). <em>shiny: Web Application
Framework for R</em>. R package version 1.9.1, <a href="https://CRAN.R-project.org/package=shiny" class="uri">https://CRAN.R-project.org/package=shiny</a>.</p></li>
<li><p>Karlsen, S. T., Rau, M. H., Sánchez, B. J., Jensen, K., &amp;
Zeidan, A. A. (2023). From genotype to phenotype: Computational
approaches for inferring microbial traits relevant to the food industry.
FEMS Microbiology Reviews, 47(4), fuad030. <a href="https://doi.org/10.1093/femsre/fuad030" class="uri">https://doi.org/10.1093/femsre/fuad030</a></p></li>
<li><p>Lê S, Josse J, Husson F (2008). “FactoMineR: A Package for
Multivariate Analysis.” Journal of Statistical Software, 25(1), 1–18. <a href="doi:10.18637/jss.v025.i01" class="uri">doi:10.18637/jss.v025.i01</a>.</p></li>
<li><p>Lees, J. A., Galardini, M., Bentley, S. D., Weiser, J. N., &amp;
Corander, J. (2018). pyseer: A comprehensive tool for microbial
pangenome-wide association studies. Bioinformatics, 34(24), 4310–4312.
<a href="https://doi.org/10.1093/bioinformatics/bty539" class="uri">https://doi.org/10.1093/bioinformatics/bty539</a></p></li>
<li><p>Oksanen J, Simpson G, Blanchet F, Kindt R, Legendre P, Minchin P,
O’Hara R, Solymos P, Stevens M, Szoecs E, Wagner H, Barbour M, Bedward
M, Bolker B, Borcard D, Carvalho G, Chirico M, De Caceres M, Durand S,
Evangelista H, FitzJohn R, Friendly M, Furneaux B, Hannigan G, Hill M,
Lahti L, McGlinn D, Ouellette M, Ribeiro Cunha E, Smith T, Stier A, Ter
Braak C, Weedon J (2024). <em>vegan: Community Ecology Package</em>. R
package version 2.6-8, <a href="https://CRAN.R-project.org/package=vegan" class="uri">https://CRAN.R-project.org/package=vegan</a>.</p></li>
<li><p>R Core Team (2024). <em>R: A Language and Environment for
Statistical Computing</em>. R Foundation for Statistical Computing,
Vienna, Austria. <a href="https://www.R-project.org/" class="uri">https://www.R-project.org/</a>.</p></li>
<li><p>Sievert C, Cheng J, Aden-Buie G (2024). <em>bslib: Custom
‘Bootstrap’ ‘Sass’ Themes for ‘shiny’ and ‘rmarkdown’</em>. R package
version 0.8.0, <a href="https://CRAN.R-project.org/package=bslib" class="uri">https://CRAN.R-project.org/package=bslib</a>.</p></li>
<li><p>Wickham H (2016). ggplot2: Elegant Graphics for Data Analysis.
Springer-Verlag New York. ISBN 978-3-319-24277-4, <a href="https://ggplot2.tidyverse.org" class="uri">https://ggplot2.tidyverse.org</a>.</p></li>
</ul>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="co"># sessionInfo()</span></span></code></pre></div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
