


<div id="welcome-to-bacpossr" class="section level2">
<h2>Welcome to BacPoSSR!</h2>
<p>For those that are not accustomed to using R packages in code format,
this app should give you an interactive idea of what BacPoSSR can
do.</p>
<div id="how-it-works" class="section level3">
<h3>How It Works</h3>
<ol style="list-style-type: decimal">
<li><strong>Select the filtering method</strong>
<ul>
<li>BacPoSSR provides two different methods to filter features from a
feature matrix. Select the tab with the method you would like to
use.</li>
<li><strong>Procrustes Analysis</strong> : In this option, your feature
matrix will be filtered based on the Procrustes distance assigned to
each feature, computed by Procrustes Analysis on the feature matrix
against its phenotype data. This uses the
<code>filterProcAnalysis</code> function in the BacPoSSR package.</li>
<li><strong>Genetic Relatedness</strong> : In this option, your feature
matrix will be filtered based on the genetic relatedness matrix computed
from the input. This option is recommended for users using Single
Nucleotide Polymorphisms (SNPs) as their features. This uses the
<code>filterGenRelatedness</code> function in the BacPoSSR package.</li>
</ul></li>
<li><strong>Upload your files and set the parameters</strong>
<ul>
<li>Please review the Input Specifications below for details on the
input data requirements and meanings.</li>
</ul></li>
<li><strong>Examine the efficacy of the filtering</strong>
<ul>
<li>After satisfying the Input Specifications, click on the “Run”
button.</li>
<li>Two plots should appear on the right side of your screen. The first
plot, is the Multiple Component Analysis (MCA) of your unfiltered, input
dataset. The second plot is the MCA of your filtered dataset. Assuming
the groups assigned to each sample are population groups, if the second
plot has less color segregation than the first, the filtering can be
said to be effective.</li>
<li>If you find that the filtering was not effective enough, try
changing the threshold parameter or consider using a different filtering
method.</li>
</ul></li>
<li><strong>Download the filtered data</strong>
<ul>
<li>If you are satisfied with the filtering, you can download the
filtered feature matrix from the button under the second MCA.</li>
</ul></li>
</ol>
</div>
<div id="input-specifications" class="section level3">
<h3>Input Specifications</h3>
<ul>
<li><em>feature matrix</em> : A feature (k-mer, SNP, unitig etc)
presence absence matrix where the rows are samples and columns are
features. The first row and column should have sample / feature
identifiers. Must be in csv file format. (For Genetic Relatedness, the
user may opt to input a genetic relatedness matrix instead as long as it
is specified in the “type of data” box.)</li>
<li><em>phenotype table</em> : A two-column table of the binary
phenotypes of each sample. The first column should be the sample
identifiers and the second column should be the phenotype assignments.
The first row should be a header. Must be in csv file format.</li>
<li><em>group table</em> : A two-column table of the population group
assignments of each sample. The first column should be the sample
identifiers and the second column should be the group assignments. The
first row should be a header. Must be in csv file format.</li>
<li><em>threshold</em> (Procrustes Analysis) : The proportion of
features that should be removed by filtering. Must be a number between 0
and 1.</li>
<li><em>threshold</em> (Genetic Relatedness) : The cut-off p-value for
filtering. Must be a number between 0 and 1.</li>
<li><em>dataset name</em> : The name of the dataset you are inputting.
This will be included in the MCA plot titles.</li>
</ul>
</div>
<div id="contact-support" class="section level3">
<h3>Contact Support</h3>
<p>For more information regarding BacPoSSR, please refer to the GitHub :
<a href="https://github.com/izumiando/BacPoSSR" class="uri">https://github.com/izumiando/BacPoSSR</a> For inquiries
regarding BacPoSSR or this Shiny app, please reach out to the author,
Izumi, at <a href="mailto:izumi.ando@mail.utoronto.ca" class="email">izumi.ando@mail.utoronto.ca</a></p>
</div>
</div>
